libr/core/cmd_search.c

```diff
@@ -1319,34 +1310,12 @@ static int r_core_search_rop(RCore *core, RInterval search_itv, int opt, const c
 		}
 	}
 
-	if (!strncmp (smode, "dbg.", 4)\
-	    || !strncmp (smode, "io.sections", 11)\
-	    || prot & R_IO_EXEC) {
-		list = r_core_get_boundaries_prot (core, prot, NULL, "search");
-	} else {
-		list = NULL;
-	}
-
-	if (!list) {
-		map = R_NEW0 (RIOMap);
-		if (!map) {
-			eprintf ("Cannot allocate map\n");
-			result = false;
-			goto bad;
-		}
-		map->fd = r_io_fd_get_current (core->io);
-		map->itv.addr = from;
-		map->itv.size = to - from;
-		list = r_list_newf (free);
-		r_list_append (list, map);
-	}
-
```

Comment:
Basically I worry that I dont understand this giant block here and below to know if Ive replaced its functionality properly. If it looks good to you all then fine by me.